timesync/peersync/sync.go

```diff
@@ -259,28 +256,27 @@ func (s *Sync) GetOffset(ctx context.Context, id uint64, prs []p2p.Peer) (time.D
 	)
 	buf, err := codec.Encode(&Request{ID: id})
```

Comment:
NIT: since we do not expect this to fail, we could use this instead and then inline `buf`

```suggestion
	buf := codec.MustEncode(&Request{ID: id})
```